### PR TITLE
Basic ansible roles and tasks

### LINKED ISSUE #3: Draft Ansible basic Ansible roles and tasks
Let's start with the basics
A webserver role for the GeoServer machines to
- [x] Install OpenJDK 8
- [x] Install Apache Tomcat 8
- [x] Install a vanilla GeoServer WAR into Tomcat (http://geoserver.org/release/stable/)
- [x] Install required utilities to mount an SMB share

A loadbalancer role for the HAProxy machines to
- [x] Install HAProxy

### PR DESCRIPTION
**Pull request by giacomoc92**
Fixes #3 

The current configuration includes one playbook for loadbalancer hosts and one for webservers, each running its own main role and the baseline.

Webserver installs a basic geoserver on top of tomcat, while loadbalancer installs haproxy and configures it to run in front of the webserver machine.

### CODE REVIEW

File: Ansible/roles/loadbalancer/templates/haproxy.cfg.j2 [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	errorfile 502 /etc/haproxy/errors/502.http
+	errorfile 503 /etc/haproxy/errors/503.http
+	errorfile 504 /etc/haproxy/errors/504.http
+
+frontend http-in
+	bind *:80
+	default_backend servers
+
+backend servers
+    server server1 {{backend}}:8080 maxconn 32
```
Comment (randomorder) [inline Ansible/roles/loadbalancer/templates/haproxy.cfg.j2]: are we using only one backend or two?

File: Ansible/roles/webserver/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  - "{{openjdk_package_name}}"
+  - unzip
+
+- name: Install utilities for SMB share
+  apt:
+    name: cifs-utils
+    state: present
+
+# tomcat installation
+- name: "add group {{tomcat_group}}"
```
Comment (randomorder) [inline Ansible/roles/webserver/tasks/main.yml]: Tomcat user and group don't need to be created. The package installer for Tomcat will take care of setting them up

File: Ansible/webServer.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+- name: fao first run
```
Comment (randomorder) [inline Ansible/webServer.yml]: "fao first run" ?

File: Ansible/inventory.ini [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[haproxy]
+51.89.118.212
+
+[geoserver]
```
Comment (randomorder) [inline Ansible/inventory.ini]: Only one GeoServer backend?

Comment (giacomoc92) [reply Ansible/inventory.ini]: At the time of the PR it was running with just one server, a second one was added afterwards. I will push the updated version

File: Ansible/roles/loadbalancer/templates/haproxy.cfg.j2 (hunk 1 above)
Comment (giacomoc92) [reply Ansible/roles/loadbalancer/templates/haproxy.cfg.j2]: ref: https://github.com/geosolutions-it/C106-2017/pull/5/files#r344634441

File: Ansible/roles/webserver/tasks/main.yml (hunk 2 above)
Comment (giacomoc92) [reply Ansible/roles/webserver/tasks/main.yml]: Going to remove them with the next commit

File: Ansible/webServer.yml (hunk 3 above)
Comment (giacomoc92) [reply Ansible/webServer.yml]: This was a temporary name for the first draft

**Pull request merged**